src/main/java/InputParser.java

```diff
@@ -0,0 +1,111 @@
+import enums.Commands;
+import enums.Errors;
+
+public abstract class InputParser {
+    // constant flags
+    private static final String LIST = "list";
+    private static final String BYE = "bye";
+    private static final String TODO = "todo";
+    private static final String EVENT = "event";
+    private static final String DELETE = "delete";
+    private static final String DONE = "done";
+    private static final String DEADLINE = "deadline";
+    private static final String BY= "/by";
+    private static final String AT = "/at";
+
+    // isolate out task name from user input
+    public static String getTaskName(String input) {
+        return input.substring(input.indexOf(" ") + 1, input.indexOf("/")).trim();
```

Comment:
These methods could perhaps follow the code quality principles:

1. [Avoid Complicated Expressions](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-complicated-expressions) - could split up instead of nesting.
2. [Avoid Magic Numbers](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-magic-numbers) - the numbers could perhaps be constants at the top of the class.
3. [Minimise Code Duplication](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#minimise-code-duplication) - there seems to be slight repetition of some code